src/auto-evo/mutation_strategy/IMutationStrategy.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Collections.Generic;
+
+public interface IMutationStrategy<T>
+    where T : Species
+{
+    public bool Repeatable { get; }
+
+    public List<Tuple<T, float>> MutationsOf(T baseSpecies, float mp);
```

Comment:
To avoid memory allocations I think it would be maybe preferable to change the signature of this... Maybe something like `public T? GenerateMutation(T baseSpecies, float mp, Func<T, bool> evaluator)` that could evaluate things one by one (of course the delegate would need some way to cache it to not create too many of them, but this could allow much more efficient use and re-using species objects as well (I'll still need to dig into some of the core classes here to make more concrete suggestions).